File: 2012/plugins_c_samples/c_only/plugins/tt.c
//
// plugins/tt.c: <tt> plugin. Looks for :tt:`text` roles, and converts
// them to <tt>text</tt>.
//
// Eli Bendersky (eliben@gmail.com)
// This code is in the public domain
//
#include "clib/dstring.h"
#include "db.h"
#include "plugin_manager.h"


static dstring tt_role_hook(dstring str, DB* db, Post* post) {
    return dstring_format("<tt>%s</tt>", dstring_cstr(str));
}


int init_tt(PluginManager* pm) {
    dstring rolename = dstring_new("tt");
    PluginManager_register_role_hook(pm, rolename, tt_role_hook);
    dstring_free(rolename);
    return 1;
}

File: 2026/plugin-pluggy/plugins/tt/tt.py
#-------------------------------------------------------------------------------
# htmlize: plugin/tt.py
#
# A htmlize plugin that handles the "tt" role.
#
# Eli Bendersky (eliben@gmail.com)
# This code is in the public domain
#-------------------------------------------------------------------------------
import htmlize


@htmlize.hookimpl
def htmlize_role_handler(role_name):
    def hook(contents):
        return f'<tt>{contents}</tt>'

    if role_name == 'tt':
        return hook
    else:
        return None
